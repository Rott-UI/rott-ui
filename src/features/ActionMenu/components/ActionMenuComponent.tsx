// React Imports
import React, {FC} from 'react'

// Component Imports
import {Item, Pressable, Label, Button, List, ActionMenu} from '../..'
import {ActionMenuStyles} from '../styles'
import {ActionMenuProps, ActionModel} from '../models'

// Constant Imports

// Util and Lib Imports

/**
 * Action Menu
 *
 * @param title Başlık alanı
 * @param subTitle Alt başlık alanı
 * @param data ActionModel[] formatında veri listesi
 * @param visible ActionMenu komponentin bulunduğu yerdeki Show state i
 * @returns Action Menu Renderlanır
 *
 */
export const ActionMenuComponent: FC<ActionMenuProps> = ({
  title,
  subTitle,
  data,
  itemHeight,
  maxItem,
  separatorTotalHeight,
}) => {
  const itemCount = (data?.length ?? 0) > maxItem! ? maxItem! : data?.length ?? 0

  // Listede renderlanacak Item
  const renderItem = ({
    title: renderItemTitle,
    name: renderItemName,
    action: onAction,
  }: ActionModel) => (
    <Pressable
      testID={`action-menu-data-${renderItemName}-test-id`}
      height={itemHeight}
      justifyContentCenter
      alignItemsCenter
      onPress={() => onAction()}>
      <Label fontSize='lg' variant='grey-900' fontWeight={500}>
        {renderItemTitle}
      </Label>
    </Pressable>
  )

  return (
    <Item size='full' style={ActionMenuStyles().actionMenuContainer} alignItemsCenter>
      <Item
        width={342}
        backgroundColor={COLOURS.GREY100}
        borderTopStartRadius={!(title || subTitle) ? 12 : 0}
        borderTopEndRadius={!(title || subTitle) ? 12 : 0}
        borderBottomStartRadius={12}
        borderBottomEndRadius={12}
        marginBottom={16}>
        <List
          testID='action-menu-test-id'
          style={{minHeight: 2}}
          width={342}
          height={itemHeight! * itemCount + separatorTotalHeight!}
          data={data}
          renderItem={({item}) => renderItem(item)}
          horizontal={false}
          separatorVariant='neutral-alpha-200'
          renderSeparator
          headerSeparator
          footerSeparator
          estimatedItemSize={display.normalize(itemHeight!, 'height')}
          showsVerticalScrollIndicator={false}
          scrollEnabled={(data?.length ?? 0) > maxItem!}
        />
      </Item>

      <Button
        variant='primary'
        onPress={() => ActionMenu.hideActionMenu()}
        fontSize='xl'
        size='full'
        testID='action-menu-cancel-test-id'
        marginBottom={21}>
        {translator('COMMON.CANCEL')}
      </Button>
    </Item>
  )
}
