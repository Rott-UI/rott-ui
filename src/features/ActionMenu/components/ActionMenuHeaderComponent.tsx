// React Imports
import React, {FC} from 'react'

// Util and Lib Imports
import {Item, Label} from '../..'
import {themeConfig} from '../../../providers'

interface ActionMenuHeaderProps {
  title?: Nullable<string>
  subTitle?: Nullable<string>
}

export const ActionMenuHeaderComponent: FC<ActionMenuHeaderProps> = ({title, subTitle}) => {
  return (
    <Item size='full' alignItemsCenter>
      <Item testID='action-menu-header-test-id' width={342}>
        {(title || subTitle) && (
          <Item
            justifyContentCenter
            alignItemsCenter
            size='full'
            backgroundColor={themeConfig.colors['grey-100']}
            borderTopStartRadius={12}
            borderTopEndRadius={12}
            height={54}>
            {title && (
              <Label
                fontSize='xl'
                variant='grey-900'
                fontWeight={700}
                marginBottom={subTitle ? 4 : undefined}>
                {title}
              </Label>
            )}
            {subTitle && (
              <Label fontSize='sm' variant='grey-900' fontWeight={500}>
                {subTitle}
              </Label>
            )}
          </Item>
        )}
      </Item>
    </Item>
  )
}
