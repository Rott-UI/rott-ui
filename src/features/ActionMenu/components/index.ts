export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
