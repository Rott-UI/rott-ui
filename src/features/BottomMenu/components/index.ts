export * from './BottomMenu'
export * from './BottomMenuItem'
