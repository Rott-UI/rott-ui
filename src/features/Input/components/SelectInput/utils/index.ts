export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
