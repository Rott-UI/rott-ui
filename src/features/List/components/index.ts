export * from './List'
export * from './FavoriteSwipeComponent'
