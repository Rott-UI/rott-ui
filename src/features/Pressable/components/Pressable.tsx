/* eslint-disable react/no-unstable-nested-components */
// React Imports
import React, {FC, LegacyRef, Ref} from 'react'

// React Native Imports
import {
  PressableProps as RNPressableProps,
  Pressable as RNPressable,
  StyleProp,
  TextStyle,
  StyleSheet,
  View,
  Animated,
} from 'react-native'

import {Label} from '../../Label'
import {PressableStyles} from '../styles'

// Model Imports
import {CommonUiProps, Size, Variant} from '../../../models'

interface PressableProps extends CommonUiProps, RNPressableProps {
  ref?: LegacyRef<View> | Ref<View | LegacyRef<View>> | any
  row?: boolean
  text?: string
  textStyle?: StyleProp<TextStyle>
  textVariant?: Variant
  textSize?: Size
  textWeight?: 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900 | 'bold' | 'normal'
  animated?: boolean
}

export const Pressable: FC<PressableProps> = ({
  text,
  textStyle,
  textVariant = 'grey-900',
  textSize = 'md',
  textWeight,
  animated = false,
  children,
  style,
  ref,
  ...props
}) => {
  const PressableContainer: FC<PressableProps> = ({
    children: containerChildren,
    ...containerProps
  }) => {
    if (animated) {
      const AnimatedPressable = Animated.createAnimatedComponent(RNPressable)

      return <AnimatedPressable {...containerProps}>{containerChildren}</AnimatedPressable>
    } else return <RNPressable {...containerProps}>{containerChildren}</RNPressable>
  }

  return (
    <PressableContainer
      animated={animated}
      ref={ref}
      testID='pressable-test-id'
      style={StyleSheet.flatten([
        PressableStyles({
          includeAlignItems: true,
          includeJustifyContent: true,
          ...props,
        }).defaultPressableStyle,
        style,
      ])}
      {...props}>
      {text && (
        <Label style={textStyle} variant={textVariant} fontSize={textSize} fontWeight={textWeight}>
          {text}
        </Label>
      )}

      {!text && <>{children}</>}
    </PressableContainer>
  )
}
