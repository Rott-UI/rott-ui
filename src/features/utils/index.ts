export * from './commonUiTestExtension'
export * from './commonUiStyleProperties'
