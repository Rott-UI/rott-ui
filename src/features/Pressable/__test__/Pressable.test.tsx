// Component Imports
import {commonUiTestExtension} from '../../utils'
import {Pressable} from '../components/Pressable'

// Util and Lib Imports
import {render, fireEvent, act} from '../../../utils'

const testId = {
  pressableTestId: 'pressable-test-id',
  pressableLabel: 'Pressable Label',
}

describe('Pressable -> Custom Component', () => {
  it('pressable component render olmalı ve snapshot eşleşmeli', () => {
    const rendered = render(<Pressable />)

    expect(rendered).toMatchSnapshot()
  })

  it('pressable componenti label ile render olmalı', async () => {
    const {pressableTestId, pressableLabel} = testId
    const {getByTestId, getByText} = render(
      <Pressable text={pressableLabel} testID={pressableTestId} />
    )

    const pressableElement = getByTestId(pressableTestId)

    expect(pressableElement).toBeTruthy()

    const labelElement = getByText(pressableLabel)

    expect(labelElement).toBeTruthy()
  })

  it('pressable tıklanabilir olmalı', async () => {
    const {pressableTestId, pressableLabel} = testId
    const onPressMock = jest.fn()
    const {getByTestId} = render(
      <Pressable
        text={pressableLabel}
        textVariant='black'
        textSize='md'
        textWeight='bold'
        testID={pressableTestId}
        onPress={onPressMock}
      />
    )
    const pressableElement = getByTestId(pressableTestId)

    expect(pressableElement).toBeTruthy()

    await act(() => {
      fireEvent.press(pressableElement)
    })

    expect(onPressMock).toBeCalled()
  })

  it('pressable disableken tıklanmamalı', async () => {
    const {pressableTestId, pressableLabel} = testId
    const onPressMock = jest.fn()
    const {getByTestId} = render(
      <Pressable
        text={pressableLabel}
        textVariant='black'
        textSize='md'
        textWeight='bold'
        testID={pressableTestId}
        onPress={onPressMock}
        disabled
      />
    )
    const pressableElement = getByTestId(pressableTestId)

    expect(pressableElement).toBeTruthy()

    await act(() => {
      fireEvent.press(pressableElement)
    })

    expect(onPressMock).not.toBeCalled()
  })

  it('pressable componenti style ile render olmalı', () => {
    const {pressableTestId, pressableLabel} = testId
    const onPressMock = jest.fn()
    const {getByText} = render(
      <Pressable
        text={pressableLabel}
        textVariant='black'
        textSize='md'
        textWeight='bold'
        testID={pressableTestId}
        onPress={onPressMock}
      />
    )

    const textElement = getByText(pressableLabel)

    expect(textElement).toHaveStyle({fontWeight: 'bold'})
  })
})

commonUiTestExtension(<Pressable testID={testId.pressableTestId} />, testId.pressableTestId)
