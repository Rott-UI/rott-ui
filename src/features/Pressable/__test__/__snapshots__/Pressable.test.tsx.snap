// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pressable -> Custom Component pressable component render olmalı ve snapshot eşleşmeli 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animated={false}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "bottom": undefined,
        "flex": 1,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      }
    }
    testID="pressable-test-id"
  />,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "flexDirection": "column-reverse",
          "justifyContent": "flex-start",
          "top": 30,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "bottom": 40,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
]
`;
